### DISCUSSION TITLE
Metalayer, updating a set of properties

### DISCUSSION THREAD

**Discussion by svevol** [pyg-team/pytorch_geometric#2349] [Q&A]
I am using Metalayer to update the edge and node properties. There are some properties that I don’t want to update, they are constant. Does anyone have an idea how I should encode that?
Example: if we think about original paper, then mass of the ball or spring constant shoul not be updated. They would be the same at the next step.

**Comment by rusty1s** [comment]
I'm not sure if I understand you correctly, but in case you do not want to update some features, you can either
1. Just omit the specific module, i.e., in case you do not want to update global features:
```python
op = MetaLayer(EdgeModel(), NodeModel(), None)
x, edge_attr, u = op(x, edge_index, edge_attr, u, batch)
```
2. Concatenate the initial values to the outputs, i.e.:
```python
op = MetaLayer(EdgeModel(), NodeModel(), GlobalModel())
x_new, edge_attr, u = op(x, edge_index, edge_attr, u, batch)
x = torch.cat([x, x_new], dim=-1)
```
